sympy/functions/elementary/piecewise.py

```diff
@@ -1019,6 +1019,58 @@ def _eval_rewrite_as_ITE(self, *args, **kwargs):
             last = ITE(c, a, last)
         return _canonical(last)
 
+    def as_non_overlapping(self, var):
+        # Returns Piecewise function with arguments which are non overlapping
+        # Can be used for multivariable as well (as long it can be simplified)
+        # Examples:
+        # >> a = Piecewise((2, x<1-y), (4, x<-1-y),(3, True))
+        # >> a.as_non_overlapping(x+y)
+        #  Piecewise((2, x + y < 1), (3, True))
+        # >> a = Piecewise((2, x>1), (4, x>3),(3, True))
+        # >> a.as_non_overlapping(x)
+        #  Piecewise((2, x > 1), (3, True))
+        from sympy import Intersection, UniversalSet, oo
+
+        newargs = []
+        existing_interval = S.EmptySet
+        _d = Dummy('d')
+        for expr,cond in self.args:
+            # Bring the expression to a single side
+            # x>1-y+z to x-1+y-z>0
+            try:
+                rearrangecond = cond.subs([(cond.rhs, 0),
+                                           (cond.lhs, cond.lhs-cond.rhs)])
+            except:
+                rearrangecond = cond
+
+            # Simplify expression to a single dummy varible
+            dummycond = rearrangecond.subs(var, _d)
```

Comment:
why use `_d` instead of `var`

Reply:
@smichr so to convert the multivariate so a single variable, otherwise it will raise error since multivariate condition simplification is not implemented in sympy. I could have done more complex case to case basis simplification, but I felt it would affect code readability and would become redundant once someone implements the multivariate simplification in Relationals. So I implemented only for expressions simplifiable to a single variable. What do you think?

Reply:
Also currently afaik, sympy treats (x>1-y).as_set() as UniversalSet, so I did not take the risk and converted everything to a single variable.

Reply:
Oh...I see, the `x + y` -> `_d` case. Hmmm. There should be a better way. This requires too much insight from the user.

Reply:
It will also fail for something like `(x+2*y).subs(x+y,z)`

Reply:
@smichr that is why I added the if simplifiable clause to the function, I added a condition check to see whether the equation is getting simplified to a single variable (dummy) or not except in the case of True condition. I think the onus is on the `solvers.py` to get this done for multivariables, I am just trying to introduce this helper functions so that I have an easier job of implementing Hilbert Transform on a Piecewise function. Also I could not understand the 'insight from users' comment? I mean the users are not supposed to do any work and throw everything at the program.